zookeeper-server/src/main/java/org/apache/zookeeper/server/NIOServerCnxn.java

```diff
@@ -639,7 +639,7 @@ private void closeSock() {
     /**
      * Close resources associated with a sock.
      */
-    @SuppressWarnings("objectconstruction:contracts.postcondition.not.satisfied") // FP: not sure why this doesn't verify
+    @SuppressWarnings("objectconstruction:contracts.postcondition.not.satisfied") // FP: java.nio.channels.Channel#isOpen lacks "@EnsuresCalledMethodsIf(expression="this", result=false, methods={"close"})" (validated)
```

Comment:
Once https://github.com/kelloggm/object-construction-checker/pull/418 is merged, we should make sure this warning goes away.